app/models/portfolio_item.rb

```diff
@@ -25,7 +25,7 @@ def item_workflow_ref
   end
 
   def validate_workflow
-    Approval.call_workflow_api { |api| api.show_workflow(workflow_ref) }
+    Approval::Service.call(ApprovalApiClient::WorkflowApi) { |api| api.show_workflow(workflow_ref) }
```

Comment:
@eclarizio Isn't that just increasing the number of changes, once Approval adds a new API we need to add a new function call Approval.some_new_api. By passing which API to use we have a single entry into the approval client and we pass in which API to call from that approval client. If a new one gets added we just need to pass the new api in we don't need to add another function in lib/approval.rb

Reply:
I like the service refactor, I agree with @mkanoor about no longer needing to add a new method call when the Approval api exposes new functionality.

I also agree with @eclarizio about now needing to pass in the constant and the method to call.

If we simply passed in the constant and used the openapi_parser, couldn't we look up the operation_id and yield that from within the service? I believe that is what @eclarizio was getting at.

Possibly:
```
Approval::Service.call { ApprovalApiClient::WorkflowAPI }
```

Service:
```
  def self.call(&block)
  # meta
  lookup tag, lookup operationId 
 yield api call from operationId
 ...
```

Reply:
@syncrou we don't have access to any openapi-parser here, we are using a generated client gem from our catalog app. The whole purpose of having this as a service is to allow us to do the setup in one place, pass the headers (x-rh-identity, x-rh-request_id etc) so that we dont have to litter that setup code in multiple places. I am not sure what is the harm in passing the constant in, since the caller is the only one that knows which API class it wants to use and once we have done the basic setup for the api instance the caller is going to decide which functions from the API class its going to call, each function in the generated code will have different arguments which you cannot encapsulate into a generic method. e.g. look at this documentation https://github.com/ManageIQ/approval-api-client-ruby/blob/master/docs/ActionApi.md for the ActionApi.md and observe the different methods that is supports. If you look at the examples you would see that we are only interested in doing the setup till the api_instance after that the onus is on the caller on which method they call and which arguments they pass into it.

Reply:
You're right that we don't have the approval openapi spec immediately available. If we did have a pattern for grabbing it and parsing it we could lookup the named operationID.

It wouldn't be a simple change though, so for this PR it is beyond the scope of the original refactor.